frontend/packages/console-shared/src/utils/resource-utils.ts

```diff
@@ -160,21 +162,27 @@ const sortByRevision = (
 };
 
 const getAnnotation = (obj: K8sResourceKind, annotation: string): string => {
-  return _.get(obj, ['metadata', 'annotations', annotation]);
+  return obj?.metadata?.annotations?.[annotation];
 };
 
-const getDeploymentRevision = (obj: K8sResourceKind): number => {
+export const getDeploymentRevision = (obj: K8sResourceKind): number => {
   const revision = getAnnotation(obj, DEPLOYMENT_REVISION_ANNOTATION);
   return revision && parseInt(revision, 10);
 };
 
-const getDeploymentConfigVersion = (obj: K8sResourceKind): number => {
+export const getDeploymentConfigVersion = (obj: K8sResourceKind): number => {
   const version = getAnnotation(obj, DEPLOYMENT_CONFIG_LATEST_VERSION_ANNOTATION);
   return version && parseInt(version, 10);
 };
 
+export const getDeploymentConfigNameForRC = (obj: K8sResourceKind): string =>
+  obj?.metadata?.ownerReferences?.find((ref) => ref.kind === DeploymentConfigModel.kind)?.name;
+
+export const getDeploymentNameForRS = (obj: K8sResourceKind): string =>
+  obj?.metadata?.ownerReferences?.find((ref) => ref.kind === DeploymentModel.kind)?.name;
+
 const getDeploymentConfigName = (obj: K8sResourceKind): string => {
-  return _.get(obj, 'metadata.ownerReferences[0].name', null);
+  return obj?.metadata?.ownerReferences?.[0]?.name;
```

Comment:
It looks like this might be the same as `getDeploymentConfigNameForRC`? (Except it assumes the first owner reference is the DC)